src/lib/migration/finish_dialog.rb

```diff
@@ -0,0 +1,125 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC, All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+
+module Migration
+  # Display the "finished" dialog, allow rebooting the machine
+  class FinishDialog
+    include Yast::Logger
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    Yast.import "UI"
+    Yast.import "Wizard"
+    Yast.import "Label"
+
+    attr_accessor :reboot
+
+    # run the dialog
+    def self.run
+      dialog = FinishDialog.new
+      dialog.run
+    end
+
+    # constructor
+    def initialize
+      textdomain "migration"
+
+      @reboot = false
+    end
+
+    # display and run the dialog
+    # @return [Symbol] user input
+    def run
+      Yast::Wizard.SetContents(
+        # TRANSLATORS: dialog title
+        _("Migration Finished"),
+        dialog_content,
+        help,
+        # going back is not possible
+        false,
+        true
+      )
+
+      Yast::Wizard.SetNextButton(:next, Yast::Label.FinishButton)
+
+      loop do
+        ret = Yast::UI.UserInput
+
+        if ret == :next && reboot?
+          # TRANSLATORS: popup message, restart can be canceled by the [Cancel] button
+          if Yast::Popup.ContinueCancel(_("The system will be restarted in one minute."))
```

Comment:
I find this a little bit confusing. Some users could understand that the system is going to be shutdown if they do nothing and clicking in 'cancel' actually aborts that shutdown. That's even what the string for the translators seems to suggest (at least to me).

The reality is that the minute will start counting after clicking 'continue'.